src/cpp/utils/SystemInfo.hpp

```diff
@@ -82,6 +82,15 @@ class SystemInfo
     static const SystemInfo& instance()
     {
         static SystemInfo singleton;
+
+        // From ctime(3) linux man page:
+        // According to POSIX.1-2004, localtime() is required to behave as though tzset(3) was called, while
+        // localtime_r() does not have this requirement. For portable code tzset(3) should be called before
+        // localtime_r().
+#if (_POSIX_C_SOURCE >= 1) || defined(_XOPEN_SOURCE) || defined(_POSIX_SOURCE)
+        tzset();
+#endif // if (_POSIX_C_SOURCE >= 1) || defined(_XOPEN_SOURCE) || defined(_POSIX_SOURCE)
+
```

Comment:
I think it is better to have this code in the constructor of the class.

The conditional should also match the one on `SystemInfo::get_timestamp`

Reply:
Addressed in 43e8b4f